src/triage/component/postmodeling/error_analysis.py

```diff
@@ -0,0 +1,169 @@
+
+    return zip(Y, X)
+
+def _error_modeler(self,
+                  depth=None,
+                  view_plots=False,
+                  **kwargs):
+   '''
+   Model labeled errors (residuals) by the error_labeler (FPR, FNR, and
+   general residual) using a RandomForestClassifier. This function will
+   yield a plot tree for each of the label numpy arrays return by the
+   error_labeler (Y).
+   Arguments:
+       - depth: max number of tree partitions. This is passed directly to
+         the classifier.
+       - view_plot: the plot is saved to disk by default, but the
+         graphviz.Source also allow to load the object and see it in the
+         default OS image renderer
+       - **kwargs: more arguments passed to the labeler: param indicating
+         the threshold value, param_type indicating the type of threshold,
+         and the path to the ProjectStorage.
+   '''
+
+   # Get matrices from the labeler
+   zip_data = self._error_labeler(param_type = kwargs['param_type'],
+                              param = kwargs['param'],
+                              path=kwargs['path'])
+
+   # Model tree and output tree plot
+   for error_label, matrix in zip_data:
+
+       dot_path = 'error_analysis_' + \
+                  str(error_label[1]) + '_' + \
+                  str(self.model_id) + '_' + \
+                  str(kwargs['param_type']) + '@'+ \
+                  str(kwargs['param']) +  '.gv'
+
+       clf = tree.DecisionTreeClassifier(max_depth=depth)
+       clf_fit = clf.fit(matrix, error_label[0])
+       tree_viz = tree.export_graphviz(clf_fit,
+                                       out_file=None,
+                                       feature_names=matrix.columns.values,
+                                       filled=True,
+                                       rounded=True,
+                                       special_characters=True)
+       graph = graphviz.Source(tree_viz)
+       graph.render(filename=dot_path,
+                    directory='error_analysis',
+                    view=view_plots)
+
+       print(dot_path)
+
+def error_analysis(self, threshold, **kwargs):
+    '''
+    Error analysis function for ThresholdIterator objects. This function
+    have the same functionality as the _error.modeler method, but its
+    implemented for iterators, which can be a case use of this analysis.
+    If no iterator object is passed, the function will take the needed
+    arguments to run the _error_modeler.
+    Arguments:
+        - threshold: a threshold and threshold parameter combination passed
+        to the PostmodelingParamters. If multiple parameters are passed,
+        the function will iterate through them.
+        -**kwags: other arguments passed to _error_modeler
+    '''
+
+    error_modeler = partial(self._error_modeler,
+                           depth = kwargs['depth'],
+                           path = kwargs['path'],
+                           view_plots = kwargs['view_plots'])
```

Comment:
this code assumes these params will be passed – so it should probably either include them in the signature or pass the entire `kwargs` without worrying about the details of the wrapped interface.

```
def error_analysis(…, **kwargs):
    error_modeler = partial(_error_modeler, model, **kwargs)
    …
```

(Among other things, with that setup, this function doesn't take on the responsibility for managing another function's signature; and, when it's violated by the user, they get a TypeError for the bad invocation, rather than a weird KeyError.)

That said, you clearly care about the individual params, and use them distinctly -- so perhaps just keep them in the signature, clearly:

```
def error_analysis(model, threshold, *, depth, path, view_plots, param, param_type):
...
```

(in that example, those params are spelled out, but because they follow `*`, they must be specified by keywords.)